### PR TITLE
use xorm builder for models.getReviewers()

### PR DESCRIPTION
**Pull request by 6543**
as title

-> https://github.com/go-gitea/gitea/pull/19030#discussion_r821995716

### CODE REVIEW

**Comment by 6543** [comment]
... this way we could re-use the condition if needed at some point ...

and first and it should dedup code and be better redable

**Review by lunny — APPROVED** [review]

File: models/repo.go [hunk 1]
```diff
@@ -218,50 +218,44 @@ func getReviewers(ctx context.Context, repo *repo_model.Repository, doerID, post
...
+				builder.Eq{"repo_id": repo.ID}.
+					And(builder.Gte{"mode": perm.AccessModeRead}),
+			),
+		))
 
 		if repo.Owner.Type == user_model.UserTypeIndividual && repo.Owner.ID != posterID {
 			// as private *user* repos don't generate an entry in the `access` table,
 			// the owner of a private repo needs to be explicitly added.
-			users = append(users, repo.Owner)
+			cond = cond.Or(builder.Eq{"id": repo.Owner.ID})
```
Comment (zeripath) [inline models/repo.go]: Shouldn't this be:

```suggestion
			cond = cond.Or(builder.Eq{"`user`.id": repo.Owner.ID})
```

File: models/repo.go [hunk 2]
```diff
@@ -218,50 +218,44 @@ func getReviewers(ctx context.Context, repo *repo_model.Repository, doerID, post
...
-			Find(&users); err != nil {
-			return nil, err
-		}
+
+		cond = cond.And(builder.In("`user`.id",
+			builder.Select("user_id").From("access").Where(
+				builder.Eq{"repo_id": repo.ID}.
+					And(builder.Gte{"mode": perm.AccessModeRead}),
+			),
+		))
```
Comment (zeripath) [inline models/repo.go:232]: having the	`cond := builder.And(builder.Neq{"`user`.id": posterID})` moved out of this and to the start means means that this not quite the same query any more.

Comment (6543) [reply models/repo.go:232]: yes but it performs the same ... but more strict ...

**Review by noerw — COMMENTED** [review]
Readability-wise this is a huge step backwards (at least the public repo query), so I can't really approve this. Functionality-wise this looks good.

File: models/repo.go (hunk 1 above)
Comment (6543) [reply models/repo.go]: we dont need it - but we can also add it ...

File: models/repo.go [hunk 3]
```diff
@@ -218,50 +218,44 @@ func getReviewers(ctx context.Context, repo *repo_model.Repository, doerID, post
...
-				repo.ID, perm.AccessModeRead,
-				posterID).
-			Find(&users); err != nil {
-			return nil, err
-		}
+
+		cond = cond.And(builder.In("`user`.id",
+			builder.Select("user_id").From("access").Where(
+				builder.Eq{"repo_id": repo.ID}.
+					And(builder.Gte{"mode": perm.AccessModeRead}),
```
Comment (lafriks) [inline models/repo.go:230]: I think it's missing and condition neq(user_id, poster id)

**Review by lafriks — APPROVED** [review]

**Pull request merged**